### ISSUE TITLE
Add to CI: Link checks

### ISSUE THREAD

**Issue by brianamarie** [github/training-kit#487]
## Overview
**TL;DR**
We want the travis tests to run to include tests for valid links.
cc @github/services-training 

### Extra Detail
- We don't want to publish content with broken links. 🔗 
- This was discovered because we had [some broken links](https://github.com/github/training-kit/issues/484#issuecomment-285374971).
- We already use this type of test in other repositories, like [on-demand-github-pages](https://github.com/githubschool/on-demand-github-pages/blob/master/tests/test_htmlValidation.rb).

### How to Contribute
- [ ] Comment on this issue if you'd like to work/collaborate on it! 
- [ ] Fork the [repository](https://github.com/github/training-kit).
- [ ] Use the [GitHub Flow](https://guides.github.com/introduction/flow/) to make changes to your fork. [This](https://services.github.com/on-demand/intro-to-github/) is a refresher course if you're unsure about how to make a change on GitHub.
- [ ] Push your changes to your repository. 
- [ ] Submit a Pull Request
  - Base Dropdown: github/training-kit
  - Compare Dropdown: Your fork

#### Testing Locally
If you'd like to make and test changes locally (and see how they would look if merged), do the following:
- In your command line:
  - [Install Git](https://git-scm.com/)
  - [Install Ruby](https://www.ruby-lang.org/en/documentation/installation/)
  - [ ] Run `script/setup`
  - [ ] Run `script/server`.
    - When successful, the script will initiate a local server at `http://127.0.0.1:4000/on-demand`.

### Questions?
- Leave a comment on this issue! Make sure to use @ mentions if you want a specific person's attention!

**Comment by PuZZleDucK** [comment]
Running `bundle exec htmlproofer --external_only ./_site/` gets you most of the way there... only there are a few false positives I think... I'm also unsure how to "run tests" for your build (looks to be manual testing?), so not sure I can make anything happen.

**Comment by stoe** [comment]
Duplicate of #547
